### PR TITLE
Bluetooth functionality 

### PR DESCRIPTION
**Pull request by R0NAM1**
In this PR Bluetooth functionality is working (With the system directory, which will need to be uploaded somewhere and that location be committed to the specific recipe). Not much more to say. Plus a shiny new boot gif!

### CODE REVIEW

**Comment by R0NAM1** [comment]
Had to merge 'main' into 'bluetooth'.

**Comment by MagneFire** [comment]
Great job on your work on getting Bluetooth to work! I know that this can be quite the nuisance to get it to work.

That being said there does appear to be an issue with this PR as all commits that are currently in the repository seem to be duplicated here too.

If I'm not mistaken the commits to get Bluetooth to work are:
- https://github.com/AsteroidOS/meta-skipjack-hybris/pull/10/commits/f1e90d82e96e3cecd17ffcff46d76e41328fb089: _Added BlueBinder Recipies From LuneOS_: You shouldn't need this commit as these recipes are available in `meta-asteroid` too now (https://github.com/AsteroidOS/meta-asteroid/commit/d7dcacc057486b676f065f8018b1eed0a1991139, https://github.com/AsteroidOS/meta-asteroid/commit/08f0ffb6f71b94e12ee8166dcdc9cd3a7cec34e4 and https://github.com/AsteroidOS/meta-asteroid/commit/8dfe99cb68800c75a012a713f8cba4ab47531171). You can drop this commit.
- https://github.com/AsteroidOS/meta-skipjack-hybris/pull/10/commits/04e4d6f96cb230231351f0d003adc8e3a65ae962: _Update Defconfig for VHCI and fix syntax_: I'll leave a review if I find any relevant comments.
- https://github.com/AsteroidOS/meta-skipjack-hybris/pull/10/commits/3b20f9d807fb01093659d3489c2004740e61ba19: _Added bluetooth support via kernel, ble patches and custom system dir with init.rc_: I'll leave a review if I find any relevant comments.

---

At this point what I would recommend instead is to get a clean copy of the repository and apply these commits on top of that as it seems that the git history has diverged quite a bit.
The approach that I usually follow in this case is to generate `*.patch` files and apply those.

To generate patch files from your current work do the following:
- `git format-patch -1 [COMMIT HASH]`
  - `git format-patch -1 04e4d6f96cb230231351f0d003adc8e3a65ae962`: For the _Update Defconfig for VHCI and fix syntax_ commit.
  - `git format-patch -1 3b20f9d807fb01093659d3489c2004740e61ba19`: For the _Added bluetooth support via kernel, ble patches and custom system dir with init.rc_ commit.
  - Keep in mind that whenever a commit is changed (using `--amend` or rebasing) the hash changes too. Remember this when creating patch files.

Now get a fresh copy of the repository:
  - `git clone https://github.com/AsteroidOS/meta-skipjack-hybris.git`.
  - Change the `remote` so you can push to your own fork: `git remote set-url origin git@github.com:R0NAM1/meta-skipjack-hybris.git`
  - (Optionally) create a new branch: `git checkout -b bluetooth`. This will allow you to update this PR.
 
Apply the patches: 
- `git am 0001-Update-Defconfig*.patch`
- `git am 0001-Added-bluetooth*.patch`

Now your repository should be in the correct state and you can push the changes `git push -u origin bluetooth`. It should fail to push as there already exists commits (from this PR). So force push instead `git push -u origin bluetooth -f`

File: conf/machine/skipjack.conf
Comment (MagneFire) [inline conf/machine/skipjack.conf]: You shouldn't need to install `libgbinder` and `libglibutil` as they are only build time dependencies for `bluebinder`. Only `bluebinder` should be good enough.

File: convert-overrides.py
Comment (MagneFire) [inline convert-overrides.py]: Why is this file included? It doesn't seem necessary?

File: recipes-core/bluebinder/bluebinder.bb
Comment (MagneFire) [inline recipes-core/bluebinder/bluebinder.bb]: This file shouldn't be needed as it's available in `meta-asteroid`.
(I get that this is a commit to fix the syntax changed with the new Honster Yocto build, just letting you know that it's not needed here :smile:  )

File: recipes-core/libgbinder/libgbinder.bb
Comment (MagneFire) [inline recipes-core/libgbinder/libgbinder.bb]: Same as above.

File: recipes-android/android/android-system-partition_skipjack-p.bb
Comment (MagneFire) [inline recipes-android/android/android-system-partition_skipjack-p.bb]: We should probably first push this to a Dropbox first before merging this :smile:

**Comment by R0NAM1** [comment]
To resolve the commit history issue, I just created a new branch, rebased the bluetooth commits into it, and force copied that branch to bluetooth. (Delete Branch Bluetooth and branch from Bluetooth-GoodHistory -> Bluetooth)

File: recipes-android/android/android-system-partition_skipjack-p.bb
Comment (R0NAM1) [reply recipes-android/android/android-system-partition_skipjack-p.bb]: How do you want me to get it to you?

File: conf/machine/skipjack.conf
Comment (R0NAM1) [reply conf/machine/skipjack.conf]: Done!

File: convert-overrides.py
Comment (R0NAM1) [reply convert-overrides.py]: Didn't realize it was even in the repo, must have accidentally run git add -A when I had multiple files and missed it, removed now.

File: recipes-core/bluebinder/bluebinder.bb
Comment (R0NAM1) [reply recipes-core/bluebinder/bluebinder.bb]: Recipe Removed.

File: recipes-core/libgbinder/libgbinder.bb
Comment (R0NAM1) [reply recipes-core/libgbinder/libgbinder.bb]: Removed.

File: recipes-android/android/android-system-partition_skipjack-p.bb
Comment (R0NAM1) [reply recipes-android/android/android-system-partition_skipjack-p.bb]: Uploaded via dropbox, link at https://dl.dropboxusercontent.com/s/3scbpxkbxv5grtp/Skipjack-System.tar.gz.

**Comment by MagneFire** [comment]
Only one more comment. It seems you are now adding things in one commit, just to remove it in another...

**Comment by R0NAM1** [comment]
Looking at them the two commits were basically opposites, so I just went ahead and squashed them down into a single commit, don't merge yet though, it might be just my build, but there is an issue.

File: recipes-android/android/android_skipjack-o.bb
Comment (jrtberlin) [inline recipes-android/android/android_skipjack-o.bb:41]: Unnecessary empty line

**Review by jrtberlin — APPROVED** [review]
Hi, thank you for your work!
There is one empty line added for some reason, but I don't see that as a blocking issue.
@MagneFire feel free to merge.

**Comment by FlorentRevest** [comment]
I'm just a bit curious about the
```
-CONFIG_USB_LIBCOMPOSITE=y
```
change in the defconfig. Does USB work before/after this PR ? If everything works then it's fine by me but I just wanted to make sure this is not an experiment that slipped into this PR. (also if it's inconsequential for this PR, generally speaking it's good to keep things in separate commits at least, PRs at best)

Anyway, thanks! :)

**Comment by MagneFire** [comment]
> I'm just a bit curious about the
> ```
> -CONFIG_USB_LIBCOMPOSITE=y
> ```
> change in the defconfig. Does USB work before/after this PR ? If everything works then it's fine by me but I just wanted to make sure this is not an experiment that slipped into this PR. (also if it's inconsequential for this PR, generally speaking it's good to keep things in separate commits at least, PRs at best)
> 
> Anyway, thanks! :) 

I noticed this too, but I think it's nothing major because the same config option also exists a bit further down. So it's a duplicate. It was probably automatically solved by the config generator from the Linux kernel.

**Pull request merged**
